src/tests/remote.test.ts

```diff
@@ -64,4 +81,33 @@ describe('Perl debugger connection', () => {
 		assert.equal(res.ln, 7); // The first code line in test.pl is 7
 	});
 
+	it('Should be able to get loaded scripts and their source code from' + FILE_TEST_PL, async () => {
+
+		const [ server, local ] = setupDebugger(
+			conn, FILE_TEST_PL, DATA_ROOT, [], launchOptions
+		);
+
+		// Wait for result
+		const res = await server;
+
+		const loadedFiles = await conn.getLoadedFiles();
+		const modulePms = loadedFiles.filter(
+			x => x.endsWith('Module.pm')
+		);
+
+		assert.ok(
+			modulePms.length > 0,
+			'Must have loaded a `Module.pm` file'
+		);
+
+		const sourceCode = await conn.getSourceCode(modulePms[0]);
+
+		assert.ok(
+			sourceCode.indexOf('Hello module') > 0,
+			'Module.pm source code contains "Hello module"'
+		);
```

Comment:
same here `expect(sourceCode).toMatchSnapshot();`

ideally testing getting the list would be inside one test and getting the source code in another - the only reason being that it's easier to figure out where theres an issue.

I haven't made the unit tests based on stories or scenarios - just plain method testing. It has to do with the fact that we are not only testing our code but also perl/perlDB/padwalker on the different platforms.

Reply:
(that's the part I would love to move out - testing the perl output in the different platforms. Grabbing a snapshot of the output from perl db and compare it to what is expected...)

Reply:
I think it is entirely likely that the result here varies across platforms, e.g. due to different newline conventions.